File: 프로그래머스/lv2/12924. 숫자의 표현/숫자의 표현.cpp
#include <string>
#include <vector>

using namespace std;

int solution(int n)
{
    int answer = 0;
    int a = 1;
    while (n > 0)
    {
        if (n % a == 0) answer++;
        n -= a;
        a += 1;
    }
    return answer;
}

File: 프로그래머스/2/12924. 숫자의 표현/숫자의 표현.java
class Solution {

    public int solution(int n) {
        int answer = 0, start = 0;
        
        while (start < n) {
            int sum = 0;
            for (int i = start++; i < n; i++) {
                sum += i + 1;
                if (sum == n) {
                    answer++;
                    break;
                } else if (sum >= n) {
                    break;
                }
            }
        }

        return answer;
    }
}